src/Compilers/CSharp/Portable/Binder/SimpleProgramBinder.cs

```diff
@@ -49,17 +42,12 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
         protected override ImmutableArray<LocalFunctionSymbol> BuildLocalFunctions()
         {
             ArrayBuilder<LocalFunctionSymbol>? locals = null;
-            // All compilation units contribute top level declarations to this scope
```

Comment:
>All compilation units contribute top level declarations to this scope [](start = 15, length = 69)

I don't understand why this changed.  Looking at the spec and the LDM notes, it is still true that "All compilation units contribute top level declarations to this scope"

Reply:
> I don't understand why this changed.  Looking at the spec and the LDM notes, it is still true that "All compilation units contribute top level declarations to this scope". Can you please explain?

Only one unit is allowed to have statements. I think that merging across all units is not worth the complexity that this would add. Scenarios that would need merging are error scenarios.

---
In reply to: [397392696](https://github.com/dotnet/roslyn/pull/42648#discussion_r397392696) [](ancestors = 397392696)

Reply:
Got it.  That includes local functions, which are statements.

---
In reply to: [397395951](https://github.com/dotnet/roslyn/pull/42648#discussion_r397395951) [](ancestors = 397395951,397392696)

Reply:
By spec, local variables from the compilation unit containing the statements are in scope in every compilation unit, but accessing them (in other compilation units) is an error. That doesn't appear to be implemented here.

---
In reply to: [397397340](https://github.com/dotnet/roslyn/pull/42648#discussion_r397397340) [](ancestors = 397397340,397395951,397392696)

Reply:
> Can you please identify tests that verify that behavior?

Scope_01, Scope_02, for example. This PR doesn't change scoping rules. They were implemented earlier.

---
In reply to: [397401548](https://github.com/dotnet/roslyn/pull/42648#discussion_r397401548) [](ancestors = 397401548,397397340,397395951,397392696)

Reply:
None of the tests named `Scope_NN` appears to test the scenario "local functions from the compilation unit containing the statements are in scope in every compilation unit, but accessing them (in other compilation units) is an error"

---
In reply to: [397414179](https://github.com/dotnet/roslyn/pull/42648#discussion_r397414179) [](ancestors = 397414179,397401548,397397340,397395951,397392696)

Reply:
> None of the tests named Scope_NN appears to test the scenario "local functions from the compilation unit containing the statements are in scope in every compilation unit, but accessing them (in other compilation units) is an error"

How about Scope_12, Scope_13?

---
In reply to: [397428120](https://github.com/dotnet/roslyn/pull/42648#discussion_r397428120) [](ancestors = 397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
Both contain only a single compilation unit.

---
In reply to: [397429849](https://github.com/dotnet/roslyn/pull/42648#discussion_r397429849) [](ancestors = 397429849,397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
How about Scope_04, Scope_05?

---
In reply to: [397431101](https://github.com/dotnet/roslyn/pull/42648#discussion_r397431101) [](ancestors = 397431101,397429849,397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
Scope_04 has a single compilation unit.  Scope_05 has a couple of tests for it (//5 and //10).  I was asking for a test that demonstrates this aspect:  "these names shadow names that would have been available from a global using static directive" ... in another compilation unit.  For example, if compilation unit 1 contains a local function called `Max` and compilation unit 2 contains a `using static System.Math;`, it still would not be possible to use `Max(1, 2)` in compilation unit 2.

---
In reply to: [397432813](https://github.com/dotnet/roslyn/pull/42648#discussion_r397432813) [](ancestors = 397432813,397431101,397429849,397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
> I was asking for a test that demonstrates this aspect:  "these names shadow names that would have been available from a global using static directive" ... in another compilation unit

It is quite possible I do not yet have tests like this because the tests I have demonstrate shadowing in the global namespace and that comes before we look in usings. I will be happy to add tests with usings in a separate PR. They are not related to the goal of this PR. And the place you are commenting on has nothing to do with implementing scoping rules. 

---
In reply to: [397435794](https://github.com/dotnet/roslyn/pull/42648#discussion_r397435794) [](ancestors = 397435794,397432813,397431101,397429849,397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
Aleksey committed offline to adding such a test in the next PR.

---
In reply to: [397439070](https://github.com/dotnet/roslyn/pull/42648#discussion_r397439070) [](ancestors = 397439070,397435794,397432813,397431101,397429849,397428120,397414179,397401548,397397340,397395951,397392696)

Reply:
>  I will be happy to add tests with usings in a separate PR. 

Added in [https://github.com/dotnet/roslyn/pull/42751](https://github.com/dotnet/roslyn/pull/42751)

---
In reply to: [397471736](https://github.com/dotnet/roslyn/pull/42648#discussion_r397471736) [](ancestors = 397471736,397439070,397435794,397432813,397431101,397429849,397428120,397414179,397401548,397397340,397395951,397392696)